public/app/plugins/datasource/elasticsearch/query_builder.js

```diff
@@ -199,29 +228,48 @@ function (queryDef) {
   };
 
   ElasticQueryBuilder.prototype.getTermsQuery = function(queryDef) {
-    var query = {
-      "size": 0,
-      "query": {
-        "filtered": {
-          "query": {
-            "query_string": {
-              "analyze_wildcard": true,
-              "query": '$lucene_query',
-            }
-          },
-          "filter": {
-            "bool": {
-              "must": [{"range": this.getRangeFilter()}]
+    var query = {};
+    if (this.esVersion >= 5) {
```

Comment:
Yes, maybe that branching is unnecessary, I wasn't able to test in previous ES versions so it was my intention to not touch code there. 

The annotations change was obvious enough for me to understand the implications in other versions, branching could be an option though.